cli/cli.go

```diff
@@ -1,45 +1,102 @@
 package cli
 
 import (
+	"errors"
 	"fmt"
-	"github.com/shpota/goxygen/codegen"
 	"io"
 	"regexp"
 )
 
-func Start(w io.Writer, commands []string) {
+type flag struct {
+	name         string
+	options      []string
+	defaultValue string
+}
+
+func (f flag) valid(input string) bool {
+	for _, val := range f.options {
+		if input == val {
+			return true
+		}
+	}
+	return false
+}
+
+func Start(w io.Writer, commands []string, generate func(string, string)) {
```

Comment:
I would rather use an interface instead of passing a function to the Start function. You could create a Generator interface

Reply:
If I used an interface I would have to either use `projectName` and `frontend` as parameters of its `Generate()` function or encapsulate them as properties of a type that would implement the interface. The first option would be similar to passing a function (because the type implementing the interface would be stateless). The second option would require creating an instance of the type inside the `cli` package (because `projectName` and `frontend` are not known in the `main` call). This would create a link between `cli` and `codegen` package and would complicate testing.
Let me know if you have an idea on how to handle this better.

Reply:
I would go with the first approach (interface with `Generate(string, string)` ), I agree that it would be similar to passing a function but I find it cleaner, that's just a personal preference

Reply:
I would propose to keep it like this for now, we can change it later.